Add 'no-restart' and 'binary-path' properties

mcoolive opened the pull request:
### Description

Add 'no-restart' and 'binary-path' properties to give more control.

'binary-path' property aims to give more control on the installation. When 'binary_path' is defined, it is used to configure memcached services and the package installation is skipped.
- commit b2e98e6 changed how the user was created. It said: "If the user gets removed later on we should be recreating it". But it causes an issue to upgrade this cookbook while keeping memcached process up, because we cannot update the HOME dir of a user executing a program (and obviously we do not want to restart memcached service for that...) The 'binary_path' feature can be a workaround.

'no_restart' property aims to prevent restart of memcached on configuration change (default to false for compatibility). The restart is often unwanted as it empties the cache; it can be unnecessary (example: if we fix a comment...) Often we want to delay or orchestrate it, in particular with a cluster of memcached (we don't want to restart all nodes at the same time).

### Issues Resolved

- https://github.com/chef-cookbooks/memcached/issues/95

### Check List

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

tas50 commented:
@mcoolive you have a cookstyle warning on the 'or' vs || and it would be really helpful to get these new properties added to the documentation in the readme otherwise no one will know they're there to use.

mcoolive commented:
Hi,
I fixed the style issue. And I added documentation about these new properties.
Note: I did not add 'no_restart' property for RUnit. Is it an issue?
